src/transport_picoquic.cpp

```diff
@@ -808,8 +827,10 @@ PicoQuicTransport::enqueue(const TransportContextId& context_id,
 
         if (stream_cnx != ctx->second.end()) {
             metrics.enqueued_objs++;
+            // add the app data to the queue
+            // mark the stream active if it isn't already
             stream_cnx->second.tx_data->push(bytes, ttl_ms, priority);
-
+            update_datagram_ready_status(stream_cnx->second.cnx, true);
```

Comment:
If this change worked, it would mitigate the need for ```checkTxData()```. In other words, we would not need ```checkTxData()```. 

Unfortunately, this change beaks thread safety as the picoquic thread is not the same thread that enqueue is being run. Calling picoquic functions without locking. 

I tried running this branch and I do see the client consistently locking up after a little while, which is likely caused by this change.

Reply:
@TimEvens  very good point. not sure how i forgot about the single threaded nature of picoquic .. Let me update the fix

Reply:
Can confirm the locking on laps from testing today with Scott. I will say, when used on client it didn't deadlock, and the CPU usage was noticeably reduced.

Reply:
@GhostofCookie / @TimEvens how do i verify the locking issue or perhaps can you share where and how did you observe the locking issue. thanks

Reply:
@suhasHere , it's a race condition, so you have to catch it.  The way I caught it was to increase the rate the client transmits data. 

```diff
diff --git a/cmd/client.cc b/cmd/client.cc
index 893c3f4..3c96d9a 100644
--- a/cmd/client.cc
+++ b/cmd/client.cc
@@ -117,14 +117,14 @@ int main() {
   uint32_t *msg_num = (uint32_t *)&data_buf;

   while (true) {
-    for (int i = 0; i < 12; i++) {
+    for (int i = 0; i < 10; i++) {
       (*msg_num)++;
       auto data = bytes(data_buf, data_buf + sizeof(data_buf));

       client->enqueue(tcid, server.proto == TransportProtocol::UDP ? 1 : stream_id,
                         std::move(data));
     }
-    std::this_thread::sleep_for(std::chrono::milliseconds(20));
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
   }
```